tests/wasmfs/wasmfs_fetch.c

```diff
@@ -6,23 +6,87 @@
  */
 
 #include <assert.h>
+#include <dirent.h>
 #include <emscripten/emscripten.h>
 #include <emscripten/wasmfs.h>
 #include <errno.h>
-#include <fcntl.h>
-#include <signal.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <sys/stat.h>
 #include <unistd.h>
 
-static backend_t make_js_file_backend(void* arg) {
-  return wasmfs_create_js_file_backend();
+void getUrlOrigin(char* ptr, int len);
+char url_orig[1024] = {};
+
+void check_file(int fd, const char* content) {
+  assert(fd >= 0);
+  struct stat st;
+  assert(fstat(fd, &st) != -1);
+  printf("file size: %lld\n", st.st_size);
+  assert(st.st_size > 0);
+
+  char buf[st.st_size];
+  int bytes_read = read(fd, buf, st.st_size);
+  printf("read size: %d\n", bytes_read);
+  buf[bytes_read] = 0;
+  printf("'%s'\n", buf);
+  assert(strcmp(buf, content) == 0);
 }
 
-int main() {
-  backend_t backend = wasmfs_create_fetch_backend("TODO/url");
+void test_url_relative() {
+  printf("Running %s...\n", __FUNCTION__);
+
+  backend_t backend2 = wasmfs_create_fetch_backend("test.txt");
+  int fd = wasmfs_create_file("/file_rel", 0777, backend2);
+  check_file(fd, "fetch 2");
+  assert(close(fd) == 0);
+}
+
+void test_url_absolute() {
+  printf("Running %s...\n", __FUNCTION__);
+
+  assert(strlen(url_orig) != 0);
+  backend_t backend = wasmfs_create_fetch_backend(url_orig);
+  int fd = wasmfs_create_file("/test.txt", 0777, backend);
+  check_file(fd, "fetch 2");
+  assert(close(fd) == 0);
+}
+
+void test_directory_abs() {
+  backend_t backend = wasmfs_create_fetch_backend(url_orig);
```

Comment:
Do these backends ever need to get destroyed?   Since each test creates a new one, does this mean that order of the tests matter?

Reply:
I don't think there is a way to destroy a backend through the API currently. Should I add one?

The order of tests doesn't matter. I needed several backends because I wanted to test different URLs.

Reply:
No need to add a way to destroy backends. If we decide we need that, we can add it separately from this PR.